mock/api.js

```diff
@@ -1,381 +1,297 @@
-              mean: 0.1,
-              stddev: 0.1,
-              stddevpct: 1,
-              uid: 'test_measurement_id',
-              measurement_type: 'test_measurement_type',
-              measurement_idx: 0,
-              measurement_title: 'diff_measurement_title',
-              '@idx': 1,
-              name: 'sample2',
-            },
-          },
-        },
-      ],
-    },
-    aggregations: {
-      id: {
-        buckets: [
-          {
-            key: 'test_run_name',
-            type: {
-              buckets: [
-                {
-                  key: 'test_measurement_type',
-                  title: {
-                    buckets: [
-                      {
-                        key: 'test_measurement_title',
-                        uid: {
-                          buckets: [
-                            {
-                              key: 'test_measurement_id',
-                            },
-                          ],
-                        },
-                      },
-                    ],
-                  },
-                },
-              ],
-            },
-          },
+export const mockTableContents = {
+  hits: {
+    total: DEFAULT_SIZE,
+    hits: new Array(DEFAULT_SIZE).fill().map((value, index) => ({
+      _source: {
+        parent: `${casual.word}/${casual.word}/${casual.word}/${casual.word}/${casual.word}/${
+          casual.word
+        }`,
+        directory: `${casual.word}/${casual.word}/${casual.word}/${casual.word}/${casual.word}/${
+          casual.word
+        }`,
+        mtime: moment.utc() + index,
+        mode: casual.word,
+        name: casual.word,
+        ancestor_path_elements: [
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
         ],
-      },
-      name: {
-        buckets: [
+        files: [
           {
-            key: 'test_run_name',
+            name: casual.uid,
+            mtime: moment.utc() + index,
+            size: casual.integer(1, DEFAULT_SIZE),
+            mode: casual.word,
+            type: casual.word,
           },
         ],
       },
-      controller: {
-        buckets: [
-          {
-            key: 'test_controller',
+    })),
+  },
+};
+
+export const mockMappings = {
+  test_index: {
+    mappings: {
+      properties: {
+        run: {
+          properties: {
+            config: { type: 'string', index: 'not_analyzed' },
+            controller: { type: 'string', index: 'not_analyzed' },
+            date: { type: 'date', format: 'dateOptionalTime' },
+            end: { type: 'date', format: 'dateOptionalTime' },
+            id: { type: 'string', index: 'not_analyzed' },
+            iterations: { type: 'string' },
+            name: { type: 'string', index: 'not_analyzed' },
+            start: { type: 'date', format: 'dateOptionalTime' },
+            user: { type: 'string', index: 'not_analyzed' },
           },
```

Comment:
Not sure how much this'll matter for e2e testing, for for the record most of the properties are `"keyword"` now. For reference, here's an actual Postman response to the `_mappings` query:

```
{
  "drb.v6.run-data.2020-04": {
    "mappings": {
      "_meta": {
        "version": "6"
      },
      "date_detection": false,
      "properties": {
        "@generated-by": {
          "type": "keyword"
        },
        "@metadata": {
          "properties": {
            "controller_dir": {
              "type": "keyword"
            },
            "file-date": {
              "type": "date"
            },
            "file-name": {
              "type": "keyword"
            },
            "file-size": {
              "type": "long"
            },
            "md5": {
              "type": "keyword"
            },
            "pbench-agent-version": {
              "type": "keyword"
            },
            "raw_size": {
              "type": "long"
            },
            "result-prefix": {
              "type": "text"
            },
            "satellite": {
              "type": "keyword"
            },
            "tar-ball-creation-timestamp": {
              "type": "date"
            },
            "toc-prefix": {
              "type": "text"
            }
          }
        },
        "@timestamp": {
          "type": "date"
        },
        "authorization": {
          "properties": {
            "access": {
              "type": "keyword"
            },
            "owner": {
              "type": "keyword",
              "null_value": "NONE"
            }
          }
        },
        "host_tools_info": {
          "type": "nested",
          "properties": {
            "hostname": {
              "type": "keyword"
            },
            "hostname-f": {
              "type": "keyword"
            },
            "hostname-s": {
              "type": "keyword"
            },
            "label": {
              "type": "keyword"
            },
            "tools": {
              "properties": {
                "disk": {
                  "type": "text"
                },
                "haproxy-ocp": {
                  "type": "text"
                },
                "iostat": {
                  "type": "text"
                },
                "mpstat": {
                  "type": "text"
                },
                "oc": {
                  "type": "text"
                },
                "perf": {
                  "type": "text"
                },
                "pidstat": {
                  "type": "text"
                },
                "pprof": {
                  "type": "text"
                },
                "proc-interrupts": {
                  "type": "text"
                },
                "proc-vmstat": {
                  "type": "text"
                },
                "prometheus-metrics": {
                  "type": "text"
                },
                "sar": {
                  "type": "text"
                },
                "turbostat": {
                  "type": "text"
                },
                "vmstat": {
                  "type": "text"
                }
              }
            }
          }
        },
        "run": {
          "properties": {
            "config": {
              "type": "keyword"
            },
            "controller": {
              "type": "keyword"
            },
            "date": {
              "type": "date"
            },
            "end": {
              "type": "date"
            },
            "id": {
              "type": "keyword"
            },
            "iterations": {
              "type": "text"
            },
            "name": {
              "type": "keyword"
            },
            "script": {
              "type": "keyword"
            },
            "start": {
              "type": "date"
            },
            "toolsgroup": {
              "type": "keyword"
            },
            "user": {
              "type": "keyword"
            }
          }
        },
        "sosreports": {
          "type": "nested",
          "properties": {
            "hostname-f": {
              "type": "keyword"
            },
            "hostname-s": {
              "type": "keyword"
            },
            "inet": {
              "type": "nested",
              "properties": {
                "ifname": {
                  "type": "keyword"
                },
                "ipaddr": {
                  "type": "ip"
                }
              }
            },
            "inet6": {
              "type": "nested",
              "properties": {
                "ifname": {
                  "type": "keyword"
                },
                "ipaddr": {
                  "type": "keyword"
                }
              }
            },
            "md5": {
              "type": "keyword"
            },
            "name": {
              "type": "keyword"
            },
            "sosreport-error": {
              "type": "text"
            }
          }
        }
      }
    }
  }
}
```

Reply:
Interesting! I was referencing responses from the dashboard deployed in production which might be referencing an older version of Elasticsearch. For now, this doesn't cause problems because the dashboard is only concerned with seeing a string as part of the response raw value. It seems "keyword" and "text" equate to a string as part of the final raw value. Please correct me if I'm wrong here.

Reply:
Yes, in ES1, we used `"string"` for most values. In ES7, indexed values are `"keyword"` and free-search text values (we don't have that many) are `"text"`.

Indeed, the dashboard mocking only needs to worry about what the dashboard cares about, and I don't believe it really cares about the type, only field names. Just thought I'd point this out... and aware that I posted way too much of the sample for that purpose. ;-)